src/main/org/bson/ByteBuf.java

```diff
@@ -702,12 +702,14 @@ private ByteBuf putJavaString(final String string) {
   private void ensureBsonRead() {
     if (this.mode == Mode.WRITE) {
       this.buffer.flip();
+      this.mode = Mode.READ;
```

Comment:
What happens if one reads-writes-reads-writes, in both MRI and JRuby?

Reply:
In MRI, this doesn't matter since there are separate pointers for read and write positions.... In JRuby we do this whole flipping maneuver... I'm going to add some tests for this... Just wanted to see if CI passed with the current changes.